web/api/v1/api.go

```diff
@@ -790,3 +793,41 @@ func parseDuration(s string) (time.Duration, error) {
 	}
 	return 0, fmt.Errorf("cannot parse %q to a valid duration", s)
 }
+
+func init() {
+	jsoniter.RegisterTypeEncoderFunc("promql.Point", MarshalPointJSON, MarshalPointJSONIsEmpty)
+}
+
+func MarshalPointJSON(ptr unsafe.Pointer, stream *jsoniter.Stream) {
```

Comment:
@brian-brazil This marshal function seems terribly out of place. The struct is defined in another package (promql) and thats also where all the tests etc. reside. Leaving this function here means that these 2 packages have to be kept up-to-date (and that someone making changes in the promql package needs to know about testing over here). It seems that a better place to put this would be *in* the promql package (where the struct is defined).

Reply:
I didn't feel it appropriate to tie jsoniter into the promql package. This is not a struct that is going to change, as this would break all users.